### PR TITLE
Add duplication scan agentic workflow

### PR DESCRIPTION
**Pull request by asizikov**
## Summary

Adds a GitHub Agentic Workflow that scans for duplicated logic every Friday or on manual dispatch, groups findings into atomic implementation issues, and assigns created issues to Copilot via safe outputs.

| Commit | Change |
|--------|--------|
| `ci: add duplication scan agentic workflow` | Adds the source workflow, compiled lock workflow, and gh-aw generated support files for weekly/manual duplication scans. |
| `ci: address duplication workflow review` | Uses the generated `create_issue` tool name in prompt guidance and regenerates the lockfile with gh-aw v0.68.3 while preserving generated-action pinning. |

## Validation

- `gh aw compile code-duplication-scan --validate --no-emit --no-check-update`
- `gh aw compile code-duplication-scan --validate --no-check-update`
- `npm run build`
- `npm run lint`
- `npm run test`

### CODE REVIEW

File: .github/workflows/code-duplication-scan.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+- [Commands or existing test suites the implementation agent should run.]
+
+## AI implementation notes
+[Mention key constraints, edge cases, and files likely involved. Make this detailed enough for Copilot to implement without rediscovering the entire context.]
+```
+
+## Output behavior
+
+- Create up to 8 issues per run.
+- Assign created issues to Copilot through the configured `create-issue` safe output.
```
[the remark this answers is not in the document]
Comment (asizikov) [reply .github/workflows/code-duplication-scan.md]: Fixed in 4545c5c by updating the workflow prompt to reference the callable safe-output tool name `create_issue`.

File: .github/workflows/code-duplication-scan.lock.yml [hunk 2]
```diff
@@ -0,0 +1,1053 @@
...
+      total_count: ${{ steps.missing_tool.outputs.total_count }}
+    steps:
+      - name: Setup Scripts
+        uses: github/gh-aw/actions/setup@58d1d157fbac0f1204798500faefc4f7461ebe28 # v0.45.0
+        with:
+          destination: /opt/gh-aw/actions
+      - name: Download agent output artifact
+        continue-on-error: true
+        uses: actions/download-artifact@018cc2cf5baa6db3ef3c5f8a56943fffe632ef53 # v6.0.0
+        with:
```
[the remark this answers is not in the document]
Comment (asizikov) [reply .github/workflows/code-duplication-scan.lock.yml:862]: I validated this against the gh-aw and actions/download-artifact docs. The lockfile is generated by gh-aw, so I regenerated it with gh-aw v0.68.3 rather than hand-editing generated YAML. The current compiler still emits these current-run artifact downloads without `actions: read` on the conclusion/safe_outputs jobs. This matches the download-artifact docs: explicit `github-token` with `actions:read` is required when downloading artifacts from another workflow run or repository; these steps download artifacts from the current workflow run. The activation job now includes `actions: read` where the newer compiler requires it, and the workflow validates cleanly.
